util-core/src/main/scala/com/twitter/concurrent/AsyncStream.scala

```diff
@@ -465,7 +469,11 @@ final class AsyncStream[+A] private (private val underlying: Future[LazySeq[A]])
 }
 
 object AsyncStream {
-  import LazySeq.{Cons, Empty, One}
+  private case object Empty extends AsyncStream[Nothing]
+  private case class M[A](fas: Future[AsyncStream[A]]) extends AsyncStream[A]
```

Comment:
How about `Embed`? I don't have a good theory why it should be `Embed`, but it sounds kinda good here.